OnlineStore/OnlineStore.DAL/Repositories/repositories/CaregoryRepository.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.EntityFrameworkCore;
+using OnlineStore.DAL.ApplicationDbContext;
+using OnlineStore.DAL.Repositories.interfaces;
+using OnlineStore.Domain.Entities;
+
+namespace OnlineStore.DAL.Repositories.repositories
+{
+    public class CategoryRepository : ICategoryRepository
```

Comment:
you should define all of basic crud operations with database in abstract class BaseRepository, and inherit from this class in your repositories. this will help you to avoid code duplication